GiEnJul/ClientApp/src/components/institution/InstitutionForm.tsx

```diff
@@ -1,364 +1,433 @@
-                                    value={contactPhoneNumber} 
-                                    name="cphone" 
-                                    id="kontaktperson" 
-                                    label="Telefon" 
-                                    autoComplete="tel"
-                                />
-                            </Grid>
-                        </Grid>
-                    </Grid>
-                    <Grid item>
-                        <InputValidator 
-                            viewErrorTrigger={viewErrorTrigger}
-                            validators={[isEmail, isNotNull]}
-                            setIsValids={setIsValidContactEmail} 
-                            errorMessages={['Eposten er ikke gyldig', 'Vennligst skriv inn en epost']}
-                            onChange={(e) => setContactEmail(e.target.value)} 
-                            value={contactEmail} 
-                            name="cemail" 
-                            id="kontaktepost" 
-                            label="Epost" 
-                            autoComplete="email"
-                        />
-                    </Grid>
-                </Grid>
-                <Grid item className="mx-5">
-                    <Button variant="contained" type="submit" color="primary">Send</Button>
-                    {isLoading && <div className="spinner-border" role="status">
-                        <span className="sr-only">Loading...</span>
-                    </div>}
-                </Grid>
-            </Grid>
-        </form>
-    );
-}
+	const [viewErrorTrigger, setViewErrorTrigger] = useState(0);
+
+	const [persons, setPersons] = useState<(IFormPerson | undefined)[]>([
+		{} as IFormPerson,
+	]);
+
+	const [location, setLocation] = useState("");
+	const [isValidLocation, setIsValidLocation] = useState(false);
+
+	const [dinnerRadio, setDinnerRadio] = useState("");
+	const [dinnerInput, setDinnerInput] = useState("");
+	const [isValidDinner, setIsValidDinner] = useState(false);
+
+	const [dessertRadio, setDessertRadio] = useState("");
+	const [dessertInput, setDessertInput] = useState("");
+	const [isValidDessert, setIsValidDessert] = useState(false);
+
+	const [specialNeeds, setSpecialNeeds] = useState("");
+
+	const [pid, setPid] = useState("");
+
+	const [contactName, setContactName] = useState("");
+	const [isValidContactName, setIsValidContactName] = useState(false);
+
+	const [contactPhoneNumber, setContactPhoneNumber] = useState("");
+	const [isValidContactPhoneNumber, setIsValidContactPhoneNumber] =
+		useState(false);
+
+	const [contactEmail, setContactEmail] = useState("");
+	const [isValidContactEmail, setIsValidContactEmail] = useState(false);
+
+	const [isLoading, setIsLoading] = useState(false);
+	const [alertMsg, setAlertMsg] = useState<React.ReactNode>("");
+	const [alertSeverity, setAlertSeverity] = useState<any>();
+	const [alertOpen, setAlertOpen] = useState(false);
```

Comment:
I think these properties could be combined to a type, containing all variables needed to fill the snackbar.

Reply:
I agree! But I think this should be done along with the refactoring all the states in this file.